### PR TITLE
Add special display (image) for folders

### PR DESCRIPTION
**Pull request by Xwilarg**
image
Must be set in the config file:
```
displays:
  "publish/demo":
    format: image
    displayOthers: true
    displayBack: true
```

Also add the possibility to disable README feature:
```
enabled_readme: false
```

### CODE REVIEW

File: _internal/DeerLister.php [hunk 1]
```diff
@@ -269,16 +306,27 @@ public function render(string $directory): string
...
                     $title = $parsedown->getTitle();
                 }
-                break;
+            }
+
+            if (!$displayBack && $f["name"] === "..")
+            { }
+            else if (!$displayOthers && isset($displayMode) && !$this->fileDisplays[$displayMode]->doesHandle($f["extension"]))
+            { }
+            else
```
Comment (TheIndra55) [inline _internal/DeerLister.php:315]: What are these empty blocks

File: _internal/DeerLister.php [hunk 2]
```diff
@@ -256,9 +278,24 @@ public function render(string $directory): string
 
         $title = $path == "" ? "Home" : basename($directory);
         $readme = null;
+        $filesFilter = [];
+        $displayMode = null;
+        $displayBack = true;
+        $displayOthers = true;
+
+        // Check the config to set the display mode and others options set
+        if (isset($this->config["displays"]) && array_key_exists($path, $this->config["displays"]) && array_key_exists($this->config["displays"][$path]["format"], $this->fileDisplays))
```
Comment (TheIndra55) [inline _internal/DeerLister.php:287]: We should really read our config in another way in another merge, to save us from these ugly if statements

File: _internal/config.yaml [hunk 3]
```diff
@@ -14,8 +14,19 @@ enabled_previews:
   - audio
   - markdown
 
+## Do we preview the README if one is present
+enabled_readme: true
```
Comment (TheIndra55) [inline _internal/config.yaml:18]: readme_enabled?

File: _internal/displays/FileDisplay.php [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+interface FileDisplay
+{
+    public function doesHandle(string $ext): bool;
```
Comment (TheIndra55) [inline _internal/displays/FileDisplay.php:12]: Maybe some function docs like in FilePreview.php

File: _internal/DeerLister.php (hunk 1 above)
Comment (Xwilarg) [reply _internal/DeerLister.php:315]: It's the cases where the file shouldn't be displayed, I think it's cleaner that just doing a huge if that negate all of those

File: _internal/DeerLister.php (hunk 2 above)
Comment (Xwilarg) [reply _internal/DeerLister.php:287]: Yeah we can prob do a class that manage the config file, not sure if it's the scope of this PR tho

File: _internal/config.yaml (hunk 3 above)
Comment (Xwilarg) [reply _internal/config.yaml:18]: If it's false the README isn't displayed, even if there is one

File: _internal/displays/FileDisplay.php (hunk 4 above)
Comment (Xwilarg) [reply _internal/displays/FileDisplay.php:12]: Will do

File: _internal/config.yaml (hunk 3 above)
Comment (TheIndra55) [reply _internal/config.yaml:18]: Ye I meant `readme_enabled` as key maybe

Comment (Xwilarg) [reply _internal/config.yaml:18]: it was to follow the syntax of the one before

Comment (TheIndra55) [reply _internal/config.yaml:18]: Ye the one above is the enabled previews, this one is whether the readme is enabled so `readme_enabled` probably sounds better but that's a nit.

Comment (Xwilarg) [reply _internal/config.yaml:18]: But I feel like having a line enabled_* and the next one *_enabled would be confusing

**Review by TheIndra55 — APPROVED** [review]

**Comment by TheIndra55** [comment]
One thing we might want to add (doesn't necessarily have to be in this PR) is a setting for resize proxy url for the images on the display page. Like you might upload a bunch of 1080p images and while opening the folder you don't want to load all 20 full 1080p images but just display a smaller version until you click on the image.

**Comment by Xwilarg** [comment]
Yup I agree

**Pull request merged**